2-Browsers/Week1/homework/ex1-bookList/index.js

```diff
@@ -18,8 +18,88 @@
 */
 //cspell: enable
 
+class DomElement {
+  constructor(tagType, tagId, cssClass, content) {
+    this.tagType = tagType;
+    this.tagId = tagId !== undefined ? tagId : '';
+    this.cssClass = cssClass !== undefined ? cssClass : '';
+    this.content = content !== undefined ? content : '';
+    this.child = [];
+  }
+  addContent(content) {
+    this.content = content;
+  }
+  addChild(child) {
+    if (Array.isArray(child)) {
+      child.forEach((element) => {
+        this.child.push(element);
+      });
+    } else {
+      this.child.push(child);
+    }
+  }
+  addCssClass(cssClass) {
+    this.cssClass = cssClass;
+  }
+  createDom() {
+    this.domObject = document.createElement(this.tagType);
+    if (this.cssClass !== '') {
+      this.domObject.classList.add(this.cssClass);
+    }
+    if (this.tagId !== '') {
+      this.domObject.id = this.tagId;
+    }
+    this.domObject.innerHTML = this.content;
```

Comment:
Please be aware that using `innerHTML` might lead to security risks when the content can be supplied by the user.
See: https://medium.com/@jenlindner22/the-risk-of-innerhtml-3981253fe217#:~:text=innerHTML%20today%20is%20cross%2Dsite,cookies%20and%20other%20personal%20information.

If  you actually only want to put text inside the element, you should use `textContent`